qcengine/procedures/torsiondrive.py

```diff
@@ -0,0 +1,289 @@
+import io
+from collections import defaultdict
+from contextlib import redirect_stderr, redirect_stdout
+from typing import TYPE_CHECKING, Any, Dict, List, Optional, Tuple, Union
+
+import numpy as np
+from pydantic import Field, validator
+from qcelemental.models import ComputeError, DriverEnum, FailedOperation, Molecule, ProtoModel, Provenance
+from qcelemental.models.procedures import (
+    OptimizationInput,
+    OptimizationProtocols,
+    OptimizationResult,
+    QCInputSpecification,
+)
+from qcelemental.util import which_import
+
+from ..extras import provenance_stamp
+from .model import ProcedureHarness
+
+if TYPE_CHECKING:
+    from ..config import TaskConfig
+
+
+class OptimizationSpecification(ProtoModel):
```

Comment:
glad to have these lodge at QCElemental/models. I think find a working apportionment of program/molecule/specification/top-level keywords in the grand torsiondrive schema but some tweaks may  be requested down the line.